client/task.go

```diff
@@ -146,13 +146,37 @@ type TaskInfo struct {
 
 	// runtime is the runtime name for the container, and cannot be changed.
 	runtime string
+
+	// runtimeOptions is the runtime options for the container, and it will be used
+	// as the default taskOptions when taskOptions is modified.
+	// https://github.com/containerd/containerd/issues/11568
+	runtimeOptions typeurl.Any
```

Comment:
runtimeOptions will be provided in task service. I doubt that we should add this field here.

https://github.com/containerd/containerd/blob/86d68096d2d033c0f161e869af8b69fbb05c6f7f/plugins/services/tasks/local.go#L229-L231

I think merge solution sounds good to me. Did you check this solution https://github.com/containerd/containerd/pull/11604?

Reply:
https://github.com/containerd/containerd/pull/11569/files#r2007190995

Good call. Applier on client sounds good to me.

Reply:
yup